backend/rust-bindings/src/lib.rs

```diff
@@ -25,6 +25,25 @@ impl FFIOption {
     }
 }
 
+fn handle_result<F, T>(result: *mut *mut c_char, f: F) -> bool where F: FnOnce() -> Result<T>, T: ToString {
```

Comment:
```suggestion
fn assign_result<T: ToString>(target: *mut *mut c_char, src: Result<T>) -> bool {
```

I don't see the need for taking a function; it could just take the `Result<T>`? In it's current form, the function doesn't just "handle" the result, it also creates it.

Reply:
I've changed it to only take the result, it doesn't feel as clean, but it avoids passing functions. :shrug: